deployer/src/tower_service/mod.rs

```diff
@@ -0,0 +1,160 @@
+pub mod middleware;
+
+use crate::deployment::{Built, DeploymentInfo, DeploymentManager, DeploymentState, Queued};
+use crate::persistence::Persistence;
+
+use std::fmt::Display;
+use std::future::Future;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use anyhow::anyhow;
+use futures::FutureExt;
+use hyper::body::HttpBody;
+use lazy_static::lazy_static;
+use regex::Regex;
+
+lazy_static! {
+    static ref SERVICES_SLASH_NAME_RE: Regex = Regex::new("^/services/([a-zA-|0-9_]+)$").unwrap();
+}
+
+#[derive(Clone)]
+pub struct Deployer {
+    deployment_manager: DeploymentManager,
+    persistence: Persistence,
+}
+
+impl Deployer {
+    pub async fn new() -> Self {
+        let persistence = Persistence::new().await;
+
+        let cpus = num_cpus::get();
+        let pipeline_count = (cpus + 2) / 3; // TODO: How many is suitable?
```

Comment:
I don't believe this should be necessary. If we want all threads in `tokio`'s pool to be used, don't we just have to `::spawn` the `handle_queue` bit of pipeline steps right after a successful resolve of `recv.recv()`? IOW, when a step gets a new input, spawn the handler and move on to listening to the queue again.

Reply:
Yeah I agree - have made it just a single pipeline with a 'fanning out' approach (i.e., continuously reading from channel and spawn as many tasks as necessary).